server/src/service_manager.rs

```diff
@@ -40,4 +56,11 @@ impl ServiceManager {
     pub fn remove_session(&mut self, path: &OwnedObjectPath) {
         self.sessions.remove(path);
     }
+
+    pub async fn prompt_index(&mut self) -> u32 {
```

Comment:
```suggestion
    pub async fn prompt_index(&self) -> u32 {
```

It is unfortunate you still make the same mistake :)